keying.go

```diff
@@ -0,0 +1,47 @@
+package srtp
+
+// KeyingMaterialExporter allows package SRTP to extract keying material
+type KeyingMaterialExporter interface {
+	// TODO: To match crypto/tls label should be a string.
+	ExportKeyingMaterial(label []byte, context []byte, length int) ([]byte, error)
+}
+
+// ExtractSessionKeysFromDTLS extracts the session keys from DTLS
+// This behavior is defined in RFC5764
+// https://tools.ietf.org/html/rfc5764
+func ExtractSessionKeysFromDTLS(exporter KeyingMaterialExporter, isClient bool) (SessionKeys, error) {
+	label := []byte("EXTRACTOR-dtls_srtp")
+	keyingMaterial, err := exporter.ExportKeyingMaterial(label, nil, (keyLen*2)+(saltLen*2))
+	if err != nil {
+		return SessionKeys{}, err
+	}
+
+	offset := 0
+	clientWriteKey := append([]byte{}, keyingMaterial[offset:offset+keyLen]...)
+	offset += keyLen
+
+	serverWriteKey := append([]byte{}, keyingMaterial[offset:offset+keyLen]...)
+	offset += keyLen
+
+	clientWriteKey = append(clientWriteKey, keyingMaterial[offset:offset+saltLen]...)
+	offset += saltLen
+
+	serverWriteKey = append(serverWriteKey, keyingMaterial[offset:offset+saltLen]...)
+
+	if isClient {
+		return SessionKeys{
+			LocalMasterKey:   clientWriteKey[0:16],
```

Comment:
Can you use the `keyLen` constants here? When we support more `ProtectionProfile`s will be helpful